src/ReverseProxy.ServiceFabric/ServiceDiscovery/Util/LabelsParser.cs

```diff
@@ -289,5 +305,36 @@ private static IReadOnlyList<string> SplitHosts(string hosts)
         {
             return hosts?.Split(',').Select(h => h.Trim()).Where(h => h.Length > 0).ToList();
         }
+
+        private static bool ContainsKey(string routeName, string expectedKeyName, string actualKey, out int keyNameEnd)
+        {
+            keyNameEnd = -1;
+            if (!actualKey.StartsWith(RoutesLabelsPrefix, StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            var routeNamePart = actualKey.AsSpan().Slice(RoutesLabelsPrefix.Length);
+            if (!routeNamePart.StartsWith(routeName, StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            var routeNameEnd = RoutesLabelsPrefix.Length + routeName.Length;
+            if (actualKey.Length == routeNameEnd || actualKey[routeNameEnd] != '.')
```

Comment:
Why are you going back to the actual key here rather than continuing from the routeNamePart?
```suggestion
            var keyPart = routeNamePart.Slice(routeName.Length);
            if (!keyPart.StartsWith(".", StringComparison.Ordinal))
            {
                return false;
            }
            keyPart = keyPart.Slice(1);
```

Reply:
It was kept as is from the original `IndexOf`-based version because a simple char comparison `actualKey[routeNameEnd] != '.'` looked a little bit faster than `StartsWith`. Replaced with your suggestion to make it consistent with the rest of the method.